src/include/migraphx/shape.hpp

```diff
@@ -354,9 +354,9 @@ struct MIGRAPHX_EXPORT shape
 
         std::size_t size(std::size_t n = 1) const { return sizeof(type) * n; }
 
-        auto is_integral() const { return std::is_integral<type>{}; }
-        auto is_signed() const { return std::is_signed<type>{}; }
-        auto is_unsigned() const { return std::is_unsigned<type>{}; }
+        bool is_integral() const { return std::is_integral<type>{}; }
+        bool is_signed() const { return std::is_signed<type>{}; }
+        bool is_unsigned() const { return std::is_unsigned<type>{}; }
```

Comment:
Curious why `auto` was changed to `bool` ?

Reply:
I think this is for readability/clarity. I don't see why these shouldn't resolve to anything but bool? Unless we want to use value here specified by stl?

https://en.cppreference.com/w/cpp/types/is_unsigned

Reply:
Doing `x.is_integral() != y.is_integral()` will fail to compile because they will be different types using `auto`. So I explicitly convert it to `bool` instead.